libs/partners/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -513,9 +522,16 @@ def _create_collection(
                 )
             )
         # Create the vector field, supports binary or float vectors
-        fields.append(
-            FieldSchema(self._vector_field, infer_dtype_bydata(embeddings[0]), dim=dim)
-        )
+        if isinstance(self.sparse_embedding_func, BaseSparseEmbedding):
+            fields.append(
+                FieldSchema(self._vector_field, DataType.SPARSE_FLOAT_VECTOR, dim=dim)
```

Comment:
FieldSchema(self._vector_field, DataType.SPARSE_FLOAT_VECTOR)
if we pass in arg `dim`, there will raise an error :"dim should not be specified for sparse vector field vector(0)"

Reply:
I did not see this error, but removed it anyway, as it's unneeded.